packages/@aws-cdk/aws-cloudfront/README.md

```diff
@@ -103,3 +114,118 @@ const distribution = new CloudFrontWebDistribution(this, 'MyDistribution', {
     ]
 });
 ```
+
+## Distribution API - Experimental
+
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--constructs-experimental-important.svg?style=for-the-badge)
+
+In addition to the APIs listed above, a new construct (`Distribution`) is under active development as a newer, friendlier way to define CloudFront
+distributions.
+
+### Creating a distribution
+
+CloudFront distributions deliver your content from one or more origins; an origin is the location where you store the original version of your
+content. Origins can be created from S3 buckets or a custom origin (HTTP server). Each distribution has a default behavior which applies to all
+requests to that distribution, and routes requests to a primary origin.
+
+#### From an S3 Bucket
+
+An S3 bucket can be added as an origin. If the bucket is configured as a website endpoint, the distribution can use S3 redirects and S3 custom error
+documents.
+
+```ts
+import * as cloudfront from '@aws-cdk/aws-cloudfront';
+
+// Creates a distribution for a S3 bucket.
+const myBucket = new s3.Bucket(...);
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: cloudfront.Origin.fromBucket(myBucket) },
+});
+
+// Equivalent to the above
+const myBucket = new s3.Bucket(...);
+cloudfront.Distribution.forBucket(this, 'myDist', myBucket);
```

Comment:
This is non-idiomatic. `forXxx` methods are for "union-like" pattern and as a rule of thumb should not accept scope and id. 

I would just leave this out or add something like `new cloudfront.BucketDistribution(this, "myDist", { bucket: myBucket })`

Reply:
I'll leave it out for now, and we can consider `new cloudfront.BucketDistribution` in the future.